core/src/types/errors.rs

```diff
@@ -4,21 +4,32 @@ use axum::response::{IntoResponse, Response};
 use crate::types::github::TokenNotFoundError;
 
 #[derive(Debug)]
-pub struct CoreError(pub anyhow::Error);
+pub enum CoreError {
+    Input(anyhow::Error),
+    Internal(anyhow::Error),
+}
 
 impl IntoResponse for CoreError {
     fn into_response(self) -> Response {
-        if self.0.downcast_ref::<TokenNotFoundError>().is_some() {
-            return TokenNotFoundError.into_response();
-        }
+        match self {
+            CoreError::Input(e) => (StatusCode::BAD_REQUEST, format!("{}", e)).into_response(),
+            CoreError::Internal(e) => {
+                if e.downcast_ref::<TokenNotFoundError>().is_some() {
+                    return TokenNotFoundError.into_response();
+                }
 
-        (StatusCode::BAD_REQUEST, format!("{}", self)).into_response()
+                (StatusCode::INTERNAL_SERVER_ERROR, format!("{}", e)).into_response()
+            }
+        }
     }
 }
 
 impl std::fmt::Display for CoreError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+        match self {
+            CoreError::Input(e) => write!(f, "{}", e),

```

Comment:
Worth any differentiation in the string for logs, or does the error type already show up?

Reply:
I think it's okay this way, the big differentiation is in the status code (400 vs 500). happy to go either way though